tests/apps/train/test_cost.py

```diff
@@ -324,7 +324,7 @@ def test_gradient(self, dim, n_mean, simple_embedding):
         This can be differentiated to give the derivative:
         d/dx E((s - x) ** 2) = 6 * n_mean + 2 * (1 - x).
         """
-        n_samples = 10000  # We need a lot of shots due to the high variance in the distribution
+        n_samples = 20000  # We need a lot of shots due to the high variance in the distribution
```

Comment:
This takes ~3 seconds to run on my laptop (probably longer on CI). Test failed often with only 10000 samples, and even more than 20000 would be better, and would allow for a lower tolerance below, but would take much longer to run.